PD-70: document pxctl cloud drive add (#381)
portworx/pxdocs: content/reference/CLI/cloud-drives-asg.md

## Cloud Drive operations

> If you run Portworx in ASG mode, the CLI provides a set of commands that display information about your EBS volumes.

Reviewer: nit: If Portworx is managing your cloud drives, the CLI provides

Author: > nit: If Portworx is managing your cloud drives, the CLI provides

It's updated.

## Cloud Drive operations

> If you run _Portworx_ in ASG mode, the CLI provides a set of commands that display the information about the EBS volumes. Let's get an overview of the available commands before diving in:

Reviewer 2: What's the "dive-in" here? I see the commands overview, but then the topic just ends. It seems like the overview _is_ the content for this topic. 

```suggestion
If you run Portworx in ASG mode, the CLI provides a set of commands that display information about your EBS volumes. 
```

### Cloud Drive Help

> 

Reviewer 2: We should probably add some context here, since we have it for the other topics, and it helps lend clarity: 
```suggestion
The cloud drive `--help` option displays the command options and arguments you can use with the `pxctl clouddrive` command. 
```

   --help, -h  show help
```

> //TODO: The `pxctl clouddrive add` subcommand is missing.... Probably we would want to update this output. However, I don't have access to a PX cluster at the moment.

Reviewer 2: I don't see `add` when I look with my cluster: 

```
# pxctl clouddrive --help
Manage cloud drives

Usage:
  pxctl clouddrive [flags]
  pxctl clouddrive [command]

Aliases:
  clouddrive, cd

Available Commands:
  inspect     Inspect and view all the drives of a DriveSet
  list        List all the cloud drives currently being used

Flags:
  -h, --help   help for clouddrive
...
```

Author: @nathan-px I raised the same issue myself and remember we discussed this on Slack. Maybe it is hidden on purpose. If so, should we add this section or not?

                Iops:  100
                Path:  /dev/xvdg
```

## Adding a Cloud Drive to the Portworx Pool

Follow these steps to add a cloud drive to the Portworx pool:

> 1. //TODO: maybe run `pxctl status` to show the initial capacity. Thoughts?

Reviewer 2: Sure, it's not a bad idea. It allows us to validate at the end

Author: @nathan-px OK. We need a sample output though...

> 2. Run `pxctl clouddrive add` and use the `--spec` flag to pass it the parameters of the new cloud drive:

Reviewer 2: what parameters can we pass?

Author: @nathan-px Don't now... Need access to a cluster to try it out or maybe someone could help.

     ```text
     pxctl clouddrive add --spec type=gp2,size=64,iops=100
     ```

3. To add a new cloud drive, the node must be put in maintenance mode:

    ```text
    pxctl service maintenance --enter
    ```

    ```output
    This is a disruptive operation, PX will restart in maintenance mode.
    Are you sure you want to proceed ? (Y/N): Y
    ```

    Wait for about 2 minutes and then check if the node is in maintenance mode:

    ```text
    pxctl status
    ```

    ```output
    PX is in maintenance mode. Use the service mode option to exit maintenance mode.
    Node ID: a0b87836-f115-4aa2-adbb-c9d0eb597668
    ...
    output truncated for brevity
    ...
    ```

4. Once the node is in maintenance mode, you can add a new cloud drive using something like this:

    ```text
    pxctl sv drive add -d /dev/xvdi
    ```

> 5. //TODO: should we check the rebalance status and wait for completion? Or start it manually?

Reviewer 2: I think the rebalance should start automatically. We might point to the same steps for scaling up: https://docs.portworx.com/install-with-other/operate-and-maintain/scaling/scale-up/#rebalance-the-storage-pool

Author: @nathan-px Cool, I'll add a reference to https://docs.portworx.com/install-with-other/operate-and-maintain/scaling/scale-up/#rebalance-the-storage-pool.
